kratos/utilities/auxiliar_model_part_utilities.cpp

```diff
@@ -376,6 +376,38 @@ void AuxiliarModelPartUtilities::RemoveConditionsAndBelongingsFromAllLevels(cons
 /***********************************************************************************/
 /***********************************************************************************/
 
+void AuxiliarModelPartUtilities::RemoveNodesFromSubModePartsWithoutCorrespondingEntities()
```

Comment:
The name is way misleading. Without checking the implementation what comes to my mind is that you want to remove the nodes but keeping the conditions and elements. Besides there is a typo. I'd suggest calling it `RemoveOrphanNodesFromSubModelParts`.

Reply:
Done